crates/sui-replay/src/lib.rs

```diff
@@ -343,12 +367,55 @@ pub async fn execute_replay_command(
             // TODO: clean this up
             Some((0u64, 0u64))
         }
+        ReplayToolCommand::ProfileTransaction {
+            tx_digest,
+            show_effects,
+            diag,
+            executor_version,
+            protocol_version,
+            profile_output,
+        } => {
+            let now = std::time::SystemTime::now()
+                .duration_since(std::time::UNIX_EPOCH)
+                .expect("Error getting system time")
+                .as_nanos();
+            let mut default_name = std::path::PathBuf::from(".");
+            default_name.push(format!("gas_profile_{}_{}.json", tx_digest, now));
+            let output_path = profile_output.or(Some(default_name));
```

Comment:
nit: use `or_else` to avoid computing the default if you don't need to.

```suggestion
            let output_path = profile_output.or_else(|| {
                let now = SystemTime::now()
                    .duration_since(UNIX_EPOCH)
                    .expect("Error getting system time")
                    .as_nanos();
                    
                let mut default_path = PathBuf::from(".");
                default_path.push(format!("gas_profile_{tx_digest}_{now}.json"));
                default_path
            });
```